sa/sa.go

```diff
@@ -1430,13 +1430,16 @@ func (ssa *SQLStorageAuthority) NewAuthorizations2(ctx context.Context, req *sap
 // GetAuthorization2 returns the authz2 style authorization identified by the provided ID or an error.
 // If no authorization is found matching the ID a berrors.NotFound type error is returned. This method
 // is intended to deprecate GetAuthorization.
-func (ssa *SQLStorageAuthority) GetAuthorization2(ctx context.Context, id *sapb.AuthorizationID2) (*corepb.Authorization, error) {
-	obj, err := ssa.dbMap.Get(authzModel{}, id.Id)
+func (ssa *SQLStorageAuthority) GetAuthorization2(ctx context.Context, req *sapb.AuthorizationID2) (*corepb.Authorization, error) {
+	if core.IsAnyNilOrZero(req, req.Id) {
```

Comment:
This is exactly where I'd make it an explicit `if req == nil || req.Id == 0`, since we're not trying to abstract away tons of validity checks all in the same file, we're just focused on checking exactly what we need to check right here.